Modules/posixmodule.c

```diff
@@ -1508,36 +1541,28 @@ posix_fildes_fd(int fd, int (*func)(int))
 static BOOL __stdcall
 win32_wchdir(LPCWSTR path)
 {
-    wchar_t path_buf[MAX_PATH], *new_path = path_buf;
-    int result;
+    wchar_t path_buf[MAXPATHLEN], *new_path;
```

Comment:
`MAXPATHLEN` was changed to 256 in Windows. That's a strange value considering `MAX_PATH` is 260 and buffers sometimes need to be `MAX_PATH + 1` even with legacy code. 

Anyway, I don't think performance is critical for `getcwd()` and `chdir()`. It would be simpler to merge `win32_wgetcwd_buf()` into `win32_wgetcwd()`, and change the definition to `static wchar_t *win32_wgetcwd(LPDWORD plength)`. The caller would have to free the buffer.

Reply:
> `MAXPATHLEN` was changed to 256 in Windows. That's a strange value considering `MAX_PATH` is 260 and buffers sometimes need to be `MAX_PATH + 1` even with legacy code.

`MAXPATHLEN` is indeed strange, I even opened an issue about it:  #88822.

I believe I changed it purely for consistency with other code in posixmodule.c, but this might be the case of foolish consistency. I've now switched to `MAX_PATH` instead.